### DISCUSSION TITLE
failed to set locale

### DISCUSSION THREAD

**Discussion by user799595** [Homebrew/discussions#3736] [Linux] [answered]
I am using linuxbrew without super user permissions. I know this is unsupported.

RHEL 7.9, HPC

It was working well until I tried to update it today.

With a fresh install:

```
git clone https://github.com/Homebrew/brew homebrew
eval "$(homebrew/bin/brew shellenv)"
export HOMEBREW_RELOCATE_BUILD_PREFIX=1
brew update --force --quiet
chmod -R go-w "$(brew --prefix)/share/zsh"
brew install glibc
```

It fails with

```Last 15 lines from ~/.cache/Homebrew/Logs/glibc/post_install.02.localedef:

homebrew/Cellar/glibc/2.35_1/bin/localedef
-i
en_US
-f
UTF-8
en_US.UTF-8

failed to set locale!
[error] cannot open locale definition file `en_US': No such file or directory
```

@danielnachun

**Comment by danielnachun** [comment, marked as answer]
You can't use `HOMEBREW_RELOCATE_BUILD_PREFIX=1` with `glibc` right now - it must be built from source.  Pretty much everything else will work afterward though.

**Reply by user799595** [reply]
Thanks @danielnachun

And thank you for all your hard work maintaining and supporting HPC linuxbrew!
